assets/src/blocks/TopicLink/TopicLinkEditor.js

```diff
@@ -0,0 +1,229 @@
+
+    return {
+      categoriesList,
+      imageId,
+      imageUrl,
+      imageAlt,
+      currentPostCategories,
+    };
+  }, [categoryId, customImageId, customImageFromId]);
+
+  if (!categoriesList.length) {
+    return __('Populating block\'s fields…', 'planet4-blocks-backend');
+  }
+
+  /**
+   * Sets the block's category based on available categories or current post categories.
+   * @return {Object} The selected category object.
+   */
+  const setBlockCategory = () => {
+    const postCategory = categoriesList.find(category => category.id === currentPostCategories[0]);
+    const blockCategory = categoriesList.find(category => category.id === categoryId);
+
+    let categoryName = null;
+
+    if (blockCategory) {
+      categoryName = blockCategory;
+      setAttributes({categoryId: parseInt(blockCategory.id)});
+    } else if (postCategory) {
+      categoryName = postCategory;
+      setAttributes({categoryId: parseInt(postCategory.id)});
+    } else {
+      categoryName = categoriesList[0];
+      setAttributes({categoryId: parseInt(categoriesList[0].id)});
+    }
+
+    setAttributes({categoryLink: categoryName?.link || ''});
+    setAttributes({categoryName: categoryName?.name || ''});
+    return categoryName;
+  };
+
+  /**
+   * Sets the object's focal position as a CSS-compatible value.
+   * @param {Object} focalPoints - Object containing x and y focal point values.
+   * @return {string} A default focal point.
+   */
+  const setObjectPosition = focalPoints => {
+    if (!focalPoints) {
+      return '50% 50%';
+    }
+    const floatX = parseFloat(focalPoints.x).toFixed(2);
+    const floatY = parseFloat(focalPoints.y).toFixed(2);
+    setAttributes({focal_points: `${floatX * 100}% ${floatY * 100}%`});
+  };
+
+  /**
+   * Converts a focal point string into an object with x and y properties.
+   * @param {string} focalPoints - The focal points string (e.g., "50% 50%").
+   * @return {Object} An object with x and y properties as decimal values.
+   */
+  const setFocalPoint = focalPoints => {
+    const [x, y] = focalPoints.split(' ').map(value => parseFloat(value) / 100);
+    return {x, y};
+  };
+
+  const categoryName = setBlockCategory();
+
+  // Update attributes with image data for frontend
+  setAttributes({
+    imageUrl: imageUrl || '',
+    imageAlt: imageAlt || '',
+  });
+
+  /**
+   * Renders the block preview in the editor.
+   * @return {JSX.Element} The block preview element.
+   */
+  const renderEditInPlace = () => (
+    <section className="topic-link-block">
+      <div className="background-image">
+        {imageUrl && (
+          <img
+            src={imageUrl}
+            alt={imageAlt}
+            style={{objectPosition: focal_points}}
+          />
+        )}
+      </div>
+      <div className="topic-link-content">
+        <p>{__('Learn more about', 'planet4-blocks-backend')} {categoryName?.name || ''}</p>
```

Comment:
I would rather use `sprintf` here, depending on the language the category name might be placed somewhere else in the sentence
See example [here](https://github.com/greenpeace/planet4-master-theme/blob/main/assets/src/js/Components/ArchivePicker/ArchivePickerToolbar.js#L27)